scrapy/http/response/text.py

```diff
@@ -39,20 +54,14 @@ class TextResponse(Response):
     attributes: Tuple[str, ...] = Response.attributes + ("encoding",)
 
     def __init__(self, *args: Any, **kwargs: Any):
-        self._encoding = kwargs.pop("encoding", None)
+        self._encoding: Optional[str] = kwargs.pop("encoding", None)
         self._cached_benc: Optional[str] = None
         self._cached_ubody: Optional[str] = None
         self._cached_selector: Optional[Selector] = None
         super().__init__(*args, **kwargs)
 
-    def _set_url(self, url):
-        if isinstance(url, str):
-            self._url = to_unicode(url, self.encoding)
```

Comment:
This method became no-op after Python 2 code removal. Initially it encoded py2 `unicode` to py2 `str` using `self.encoding` but now both the input and the internal representiation are str-only.